builder/parser/line_parsers.go

```diff
@@ -10,13 +10,12 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
-	"strconv"
 	"strings"
 	"unicode"
 )
 
 var (
-	errDockerfileJSONNesting = errors.New("You may not nest arrays in Dockerfile statements.")
+	errDockerfileNotStringArray = errors.New("When using JSON array syntax, arrays must be comprised of strings only.")
```

Comment:
"must be comprised of"; Perhaps something simpler, like "arrays can only contain strings / string-values". Trying to not use "comprised" here, because that may be difficult to understand for non-English users (not sure)